lib/Dialect/Arc/Transforms/StripSV.cpp

```diff
@@ -151,9 +151,19 @@ void StripSVPass::runOnOperation() {
         else
           next = reg.getNext();
 
+        Value presetValue;
+        // Materialize initial value, assume zero initialization as default.
+        if (reg.getPreset() && !reg.getPreset()->isZero()) {
+          assert(hw::type_isa<IntegerType>(reg.getType()) &&
+                 "cannot lower non integer preset");
+          presetValue = builder.createOrFold<hw::ConstantOp>(
+              reg.getLoc(), IntegerAttr::get(reg.getType(), *reg.getPreset()));
+        }
```

Comment:
I like the idea of having an implicit initialization value as a default, e.g. `0` for two-valued, `X` for four-valued, and `U` for nine-valued integers, and then only add explicit initializers if they differ. Most SV designs will not have initializers on their registers, so that will make the common case less verbose :+1: